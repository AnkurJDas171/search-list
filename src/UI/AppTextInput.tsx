import React from "react";
import { StyleSheet } from "react-native";
import { Input } from "@rneui/themed";

import { AppTextInputProps } from "./types";
import colors from "../assets/colors";
import { height } from "../utils";
import { testID } from "../assets/Constants";


const AppTextInput = ({
    handleTextChange,
    icon,
    placeholder,
    value
}: AppTextInputProps): JSX.Element => {
    return (
        <Input
            containerStyle={styles.container}
            inputContainerStyle={styles.inputContainer}
            onChangeText={handleTextChange}
            placeholder={placeholder}
            leftIcon={icon}
            testID={testID.APP_TEXT_INPUT}
            value={value}
        />
    )
}

const styles = StyleSheet.create({
    container: {
        alignItems: 'center',
        backgroundColor: colors.TEXT_INPUT,
        borderRadius: 50,
        display: 'flex',
        flexDirection: 'row',
        width: '45%',
        height: height(40)
    },
    inputContainer: {
        borderBottomWidth: 0,
    }
})

export default AppTextInput;
