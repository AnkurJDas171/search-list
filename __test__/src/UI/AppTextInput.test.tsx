import React from "react";
import { render } from "@testing-library/react-native";

import AppTextInput from "../../../src/UI/AppTextInput";
import SearchIcon from "../../../src/assets/icons/Search";

describe("Text Input", ()=>{
    it("should render text input", async () => {
        const icon = <SearchIcon height="40" width="40" />;

        const screen = render(
            <AppTextInput
                handleTextChange={()=>{}}
                icon={icon}
                placeholder={"sample text"}
            />
        )
        
        expect(screen.toJSON()).toMatchSnapshot();

        const textInput = await screen.findByTestId("app-text-input");
        expect(textInput).not.toBeNull();
    })
})
