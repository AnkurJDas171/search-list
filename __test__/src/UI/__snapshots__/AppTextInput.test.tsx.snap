// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Input should render text input 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "rgba(243, 243, 243, 1)",
      "borderRadius": 50,
      "display": "flex",
      "flexDirection": "row",
      "paddingHorizontal": 10,
      "width": "55%",
    }
  }
  testID="RNE__Input__view-wrapper"
>
  <View
    collapsable={false}
    style={
      {
        "alignItems": "center",
        "borderBottomWidth": 0,
        "borderColor": "#86939e",
        "flexDirection": "row",
        "transform": [
          {
            "translateX": 0,
          },
        ],
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "height": 40,
          "justifyContent": "center",
          "marginVertical": 4,
          "paddingRight": 4,
        }
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="40"
        bbWidth="40"
        focusable={false}
        height="40"
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "flex": 0,
              "height": 40,
              "width": 40,
            },
          ]
        }
        vbHeight={64}
        vbWidth={64}
        width="40"
        x="0px"
        xml="
    <svg xmlns="http://www.w3.org/2000/svg" x = "0px" y = "0px" width = "100" height = "100" viewBox = "0 0 64 64" >
        <path d="M 24 2.8886719 C 12.365714 2.8886719 2.8886719 12.365723 2.8886719 24 C 2.8886719 35.634277 12.365714 45.111328 24 45.111328 C 29.036552 45.111328 33.664698 43.331333 37.298828 40.373047 L 52.130859 58.953125 C 52.130859 58.953125 55.379484 59.435984 57.396484 57.333984 C 59.427484 55.215984 58.951172 52.134766 58.951172 52.134766 L 40.373047 37.298828 C 43.331332 33.664697 45.111328 29.036548 45.111328 24 C 45.111328 12.365723 35.634286 2.8886719 24 2.8886719 z M 24 7.1113281 C 33.352549 7.1113281 40.888672 14.647457 40.888672 24 C 40.888672 33.352543 33.352549 40.888672 24 40.888672 C 14.647451 40.888672 7.1113281 33.352543 7.1113281 24 C 7.1113281 14.647457 14.647451 7.1113281 24 7.1113281 z" > </path>
    </svg>
"
        xmlns="http://www.w3.org/2000/svg"
        y="0px"
      >
        <RNSVGGroup
          fill={
            {
              "payload": 4278190080,
              "type": 0,
            }
          }
        >
          <RNSVGPath
            d="M 24 2.8886719 C 12.365714 2.8886719 2.8886719 12.365723 2.8886719 24 C 2.8886719 35.634277 12.365714 45.111328 24 45.111328 C 29.036552 45.111328 33.664698 43.331333 37.298828 40.373047 L 52.130859 58.953125 C 52.130859 58.953125 55.379484 59.435984 57.396484 57.333984 C 59.427484 55.215984 58.951172 52.134766 58.951172 52.134766 L 40.373047 37.298828 C 43.331332 33.664697 45.111328 29.036548 45.111328 24 C 45.111328 12.365723 35.634286 2.8886719 24 2.8886719 z M 24 7.1113281 C 33.352549 7.1113281 40.888672 14.647457 40.888672 24 C 40.888672 33.352543 33.352549 40.888672 24 40.888672 C 14.647451 40.888672 7.1113281 33.352543 7.1113281 24 C 7.1113281 14.647457 14.647451 7.1113281 24 7.1113281 z"
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
    <TextInput
      editable={true}
      onChangeText={[Function]}
      placeholder="sample text"
      placeholderTextColor="#86939e"
      style={
        {
          "color": "#242424",
          "flex": 1,
          "fontSize": 18,
          "minHeight": 40,
        }
      }
      testID="app-text-input"
      underlineColorAndroid="transparent"
    />
  </View>
  <Text
    style={
      {
        "color": "#ff190c",
        "fontSize": 12,
        "margin": 5,
      }
    }
  />
</View>
`;
